### PR TITLE
[networks] Fix HTTP stats lookup during encoding

### PR DESCRIPTION
**Pull request by p-lambert**
### What does this PR do?

Fix HTTP stats lookup during serialization. Two issues were causing HTTP stats to be dropped:
* NAT translations were not being accounted for;
* Server-side lookups were being done using the wrong (inverted) keys;

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details that should be tested during the QA.

### CODE REVIEW

File: pkg/network/encoding/format.go [hunk 1]
```diff
@@ -176,6 +177,23 @@ func FormatHTTPStats(httpData map[http.Key]http.RequestStats) map[http.Key]model
 	return aggregationsByKey
 }
 
+func httpKeyFromConn(c network.ConnectionStats) http.Key {
```
Comment (heyronhay) [inline pkg/network/encoding/format.go:180]:
```suggestion
// Build the key for the http map based on whether the local or remote side is http.
func httpKeyFromConn(c network.ConnectionStats) http.Key {
```

**Review by heyronhay — APPROVED** [review]
(I'm trying to add comments in reviews to help me understand PRs better and make life easier on submitters rather than just tell them to add comments)

Looks good!

File: pkg/network/encoding/format.go [hunk 2]
```diff
@@ -176,6 +177,24 @@ func FormatHTTPStats(httpData map[http.Key]http.RequestStats) map[http.Key]model
...
+	raddr, rport := nat.GetRemoteAddress(c)
+
+	if http.IsHTTP(int(rport)) {
+		return http.NewKey(laddr, raddr, lport, rport, "")
+	}
+
+	if http.IsHTTP(int(lport)) {
+		// Since HTTP data is always indexed as (client, server), we flip the lookup key
+		return http.NewKey(raddr, laddr, rport, lport, "")
+	}
```
Comment (leeavital) [inline pkg/network/encoding/format.go:193]: For later: could this be changed to use direction?  I can see this code being a problem when we need to support HTTP servers on non standard ports.

Comment (p-lambert) [reply pkg/network/encoding/format.go:193]: You're right. I'll make sure to revisit this once we add support for non standard ports.

**Pull request merged**